src/js/components/TaskTable.js

```diff
@@ -157,9 +143,40 @@ class TaskTable extends React.Component {
     let statusClassName = TaskUtil.getTaskStatusClassName(task);
     let statusLabelClasses = `${statusClassName} table-cell-value`;
 
+    let {state} = task;
+
+    let dangerState = TaskStates[state].stateTypes.includes('failure');
+
+    let healthy = task.statuses.some(function (status) {
+      return status.healthy;
+    });
+
+    let unknown = task.statuses.some(function (status) {
+      return status.healthy == null;
+    });
+
+    let activeState = TaskStates[state].stateTypes.includes('active');
+
+    let running = ['TASK_RUNNING', 'TASK_STARTING'].includes(state) && unknown;
+    let success = healthy && state === 'TASK_RUNNING';
+    let danger = (dangerState && !activeState && dangerState &&
```

Comment:
Any chance we can refactor this into separate sequential conditions? Might be easier to read.